### PR TITLE
Add `cask_files` method to `Tap` class.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
👍 nice work

File: Library/Homebrew/cask/lib/hbc/cli/search.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def self.search(*arguments)
       partial_matches = Hbc::CLI.nice_listing(Hbc.all_tokens).grep(%r{#{search_regexp}}i)
     else
       # suppressing search of the font Tap is a quick hack until behavior can be made configurable
-      all_tokens = Hbc::CLI.nice_listing Hbc.all_tokens.reject { |t| %r{^caskroom/homebrew-fonts/}.match(t) }
+      all_tokens = Hbc::CLI.nice_listing Hbc.all_tokens.reject { |t| %r{^caskroom/fonts/}.match(t) }
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/search.rb]: @jawshooah, do you know why fonts are excluded from search?

**Comment by UniqMartin** [comment]
:+1:

File: Library/Homebrew/cask/lib/hbc/cli/search.rb (hunk 1 above)
Comment (jawshooah) [inline Library/Homebrew/cask/lib/hbc/cli/search.rb]: Looks like the filter was added in https://github.com/caskroom/homebrew-cask/pull/3025. Not sure why fonts would be too noisy unless you just search `font`. Probably fine to remove this.

**Comment by MikeMcQuaid** [comment]
@reitermarkus 🚢 when ready.

**Comment by reitermarkus** [comment]
Just want to do a rebase again before merging to see the coverage results.

**Pull request merged**